python-lib/dku_utils/access.py

```diff
@@ -21,19 +22,25 @@ def _is_none_or_blank(x):
 
 def _has_not_blank_property(d, k):
     return k in d and not _is_none_or_blank(d[k])
-    
+
 def _default_if_blank(x, d):
     if _is_none_or_blank(x):
         return d
     else:
         return x
-    
+
 def _default_if_property_blank(d, k, v):
     if not k in d:
         return v
     x = d[k]
     return _default_if_blank(x, v)
 
+def check_type_for_string(a, b):
+    if sys.version_info < (3, 0):
+        return (isinstance(a, str) or isinstance(a, unicode)) and (isinstance(b, str) or isinstance(b, unicode))
+    else:
+        return isinstance(a, str) and isinstance(b, str)
+
```

Comment:
I would prefer something like:
```python
def _is_string_type(a):
    if sys.version_info < (3, 0):
        return isinstance(a, basestring)
    else
        return isinstance(a, str)
```
(Note: `basestring` is the common ancestor of `unicode` and `str` in python2, but has been removed in python3 since there is only `str`now)


and then you can keep the logic in the `_merge_objects` function:
```python
elif _is_string_type(a) and _is_string_type(b):
    return b
```

Reply:
or use something like what is in the migration script, it's even easier that way:
```python
if sys.version_info > (3,):
    dku_basestring_type = str
else:
    dku_basestring_type = basestring
```
and then just use `isinstance(a, dku_basestring_type)`